messaging/src/main/java/org/axonframework/common/property/AbstractMethodPropertyAccessStrategy.java

```diff
@@ -37,20 +39,23 @@ public abstract class AbstractMethodPropertyAccessStrategy extends PropertyAcces
     @Override
     public <T> Property<T> propertyFor(Class<? extends T> targetClass, String property) {
         String methodName = getterName(property);
-        try {
-            final Method method = targetClass.getMethod(methodName);
-            if (!Void.TYPE.equals(method.getReturnType())) {
-                return new MethodAccessedProperty<>(method, property);
-            }
-            logger.debug(
-                    "Method with name '{}' in '{}' cannot be accepted as a property accessor, as it returns void",
-                    methodName, targetClass.getName());
-        } catch (NoSuchMethodException e) {
+        Optional<Method> method = getMethod(targetClass, methodName);
+        if (!method.isPresent()) {
             logger.debug("No method with name '{}' found in {} to use as property accessor. " +
                                  "Attempting to fall back to other strategies.",
                          methodName, targetClass.getName());
+            return null;
+        } else {
+            return new MethodAccessedProperty<>(method.get(), property);
         }
-        return null;
+    }
+
+    private <T> Optional<Method> getMethod(Class<T> targetClass, String methodName) {
+        return Arrays.stream(targetClass.getMethods())
+                     .filter(method -> method.getName().equals(methodName))
+                     .filter(method -> method.getParameterCount() == 0)
```

Comment:
Because the getter method never has parameters, if it follows either the bean or the uniform standard. In the original code also the method without arguments was requested. I am not sure what the result would have been in the old code, if the method has arguments.